src/main/kotlin/com/mineinabyss/looty/features/recipes/ItemRecipeSystem.kt

```diff
@@ -10,36 +11,34 @@ import com.mineinabyss.looty.config.looty
 import org.bukkit.Bukkit
 import org.bukkit.NamespacedKey
 import org.bukkit.event.EventHandler
+import org.bukkit.event.EventPriority
 import org.bukkit.event.Listener
+import org.bukkit.event.inventory.PrepareItemCraftEvent
 import org.bukkit.event.player.PlayerJoinEvent
 import org.bukkit.inventory.ItemStack
 
-class ItemRecipeQuery : Query() {
-    private val TargetScope.setRecipes by get<SetRecipes>()
+class ItemRecipeQuery : Query(), Listener {
+    private val TargetScope.recipes by get<SetRecipes>()
     private val TargetScope.prefabKey by get<PrefabKey>()
 
     fun TargetScope.registerRecipes(): Set<NamespacedKey> {
         val discoveredRecipes = mutableSetOf<NamespacedKey>()
-        val result: ItemStack? = setRecipes.result?.toItemStackOrNull() ?: gearyItems.createItem(prefabKey)
+        val result: ItemStack? = recipes.result?.toItemStackOrNull() ?: gearyItems.createItem(prefabKey)
 
         if (result == null) {
             looty.plugin.logger.warning("Recipe ${prefabKey.key} is missing result item")
             return emptySet()
         }
 
-        setRecipes.removeRecipes.forEach {
-            runCatching {
```

Comment:
You lost the runCatching from a previous commit here, i think the same commit also changed recipes -> setRecipes and stopped extending Listener, in which case you should the new event handler below into a separate listener